crates/containerd-shim-wasm/src/container/engine.rs

```diff
@@ -62,7 +65,7 @@ pub trait Engine: Clone + Send + Sync + 'static {
     /// The cached, precompiled layers will be reloaded on subsequent runs.
     /// The runtime is expected to return the same number of layers passed in, if the layer cannot be precompiled it should return `None` for that layer.
     /// In some edge cases it is possible that the layers may already be precompiled and None should be returned in this case.
-    fn precompile(&self, _layers: &[WasmLayer]) -> Result<Vec<Option<Vec<u8>>>> {
+    async fn precompile(&self, _layers: &[WasmLayer]) -> Result<Vec<PrecompiledLayer>> {
```

Comment:
Since this now provides a way to do more than just pre-compiling such as composing I wonder if we would want to rename it

Reply:
Yeah that definitely seems reasonable! Happy to call this `process_layers` or whatever makes sense.

Reply:
Should `can_precompile()` evolve as well? It seems a little strange to rename one and not the other. Additionally there are various labels applied with a `*.precompile` segment; should we rename those to `process` as well to maintain consistency?